### ISSUE TITLE
There is an Error Syntax on functtest.py and login failed after disabled the method that make this error

### ISSUE THREAD

**Issue by jono123** [jaw20/Crunchyroll-XML-Decoder#11]
-- Fresh install python 2.7.9
-- pip install m3u8 and cfscrape

--The Result is : 

Traceback (most recent call last):
  File "D:\New folder (2)\Crunchyroll-XML-Decoder-master\Crunchyroll-XML-Decoder
-master\crunchy-xml-decoder.py", line 9, in <module>
    import functtest
  File "crunchy-xml-decoder\functtest.py", line 98
    except KeyError:
         ^
SyntaxError: invalid syntax
Press any key to continue . . .

--- disable method to check whether cfscrape are installed or not. (line 89 - 99 in functtest.py).
And the result is :

python version=2.7.9 32 bit
OS Version=Windows 8 6.2.9200
System Type=AMD64
Crypto installed
lxml installed
Do you have an account [Y/N]?y
Username: biasa123
Password(don't worry the password are typing but hidden:
Login failed.
Press any key to continue . . .

**Comment by OhNiice** [comment]
Display your non-printable characters!

In the "functtest.py" file, both lines 98 & 99 are indented with tabs. 
Just reindent them with spaces and you're done. 

That's one of the reasons I don't like python...

**Comment by albogango** [comment]
I'm experiencing the same problem.

Also, I'm unable to open the "functtest.py" file. It closes instantly.

**Comment by OhNiice** [comment]
Indeed, you're not supposed to execute this file directly. There is no "main" so executing it will result in an instant close.

You have to execute the .bat or .py file in the base directory to execute the whole thing.

--

I changed multiple things with no success... I didn't dig deeply though.

________________________________

**Comment by jono123** [comment]
how to check if login.py success to bypass cloudflare?

if the old configuration in login.py can login properly before CR use the cloudflare then scrape function may not work properly.

**Comment by albogango** [comment]
I see. Could you give me step-by-step instructions in layman's terms? I don't understand what you mean by executing the .bat or .py file in the base directory.

**Comment by OhNiice** [comment]
@jono123 I think "login.py" successes to bypass Cloudflare but the data currently sent for the connection isn't enough anymore.

@albogango The file is the "crunchy-xml-decoder.bat" or ".py" file. You can find in the project's base directory. You can find step-by-step instructions in the [README.md file](https://github.com/jaw20/Crunchyroll-XML-Decoder/blob/master/README.md).

**Comment by albogango** [comment]
@OhNiice That was my original issue, like the topic poster. I suppose my question would be, how do I reindent spaces in lines 98 & 99 in the functtest.py file?

**Comment by OhNiice** [comment]
@albogango Oh, OK! 

You need to open "login.py" with a text editor like notepad++ and display the non-printable characters (look for [this button](http://puu.sh/tVB77/5e2aedebfa.png)). Then go lines 98 & 99 and replace the tab characters with spaces (1 tab character = 4 spaces, 2 tabs = 8 spaces).

**Comment by albogango** [comment]
Worked like a charm. Thanks very much, @OhNiice. Now, it seems that the only problem is one that appeared with previous versions of the Decoder. That is, it downloads as FLV with smaller file size and bitrate.

**Comment by jono123** [comment]
@albogango Can you login? What version did you use?
smaller size because resolution? 360p or 480p?
Download as flv? maybe the video doesnot muxed to mkv.

If only there is a guest mode that can directly download video or sub without login first like the old darkstar version.

**Comment by albogango** [comment]
@jono123 I was able to log in, and, actually, it did download the video as mkv with full 1080p resolution. However, it's a considerably smaller file size and lower bitrate than when downloaded with previous versions of the decoder.

Here's the comparison:

Toriko episode 90 (using jaw20's Crunchyroll-XML-Decoder)
-298 MB, 1920x1080 resolution, frame rate: 29.970000

Toriko episode 95 (using einstein95's Crunchyroll-XML-Decoder)
-537 MB, 1920x1080 resolution, frame rate: 59.000531

The only reason I didn't continue with the previous decoder is that, for whatever reason, it's not letting me download episode 90.

**Comment by jono123** [comment]
what python version did you have? 

and can you login with einstein95's version at right now? The einstein95's version doesnot have cfscrape function to bypass the cloudfare

**Comment by albogango** [comment]
@jono123 I'm using 2.7.9.

I'm able to log in to both versions. For the most part, einstein95's version works perfectly, but every now and again, a certain episode doesn't download. I believe this to be a totally random occurrence, and I don't think it follows any particular pattern. For example, in my case, I can't download Saint Seiya episode 14, and Toriko episode 90.

**Comment by jono123** [comment]
Now i can login after make a change on line 62 in login.py
from if username != '' and userstatus[1] == 'Guest':
to if username == '':

The results is ' Login as guest succesfully' . But it created a cookies.
When i look at the cookies some value is same as the cookies value from my browser.

I think i cannot login before because status on getuserstatus always return the value with 'guest'.
 
@albogango Same here. WHY?
Because : 
- When using jaw20's Crunchyroll-xml-decoder, after booting up.... text it show '1/143<-[2k'
I look at export folder the video that downloaded have .ts (when the download is running) as the extension. And the video was using 'CRF profile (larger than 22)' on x264 encoding settings and that why the size is smaller.

-When using einstein95's Crunchyroll-xml-decoder, after booting up... text it show a media info about the video and then 1180.0 kB / 2.6 sec (0.1 %)
I look at export folder the video that downloaded have .flv as the extension. The video was using '2 pass bitrate' profile on x264 encoding settings.  And we can estimate the size of the video.

Look like crunchyroll has 2 tipe of video. If only the crf version video use a smaller value than 19 i think it is good.

...note : I can download toriko episode 90 on both version with different size because different encoding profile.

**Comment by albogango** [comment]
@jono123 Yes, that's exactly it.

Has that ever happened for you? The decoder unable to download select episodes?
